orcestra_book/plans/HALO_20240919a_plan.py

```diff
@@ -0,0 +1,125 @@
+''' helper functions for rendering HALO-20240919a flight plan'''
+
+def geod_azi(latlon1, latlon2):
+    import pyproj
+    geod = pyproj.Geod(ellps="WGS84")
+
+    return geod.inv(latlon1.lon, latlon1.lat, latlon2.lon, latlon2.lat)[0]
+
+def geod_dist(latlon1, latlon2):
+    import pyproj
+    geod = pyproj.Geod(ellps="WGS84")
+
+    return geod.inv(latlon1.lon, latlon1.lat, latlon2.lon, latlon2.lat)[2]
+
+def centre_from_tangent(latlon1, latlon2, radius, add_angle=0):
+    import pyproj
+    geod = pyproj.Geod(ellps="WGS84")
+ 
+    az12, az21, dist = geod.inv(latlon1.lon, latlon1.lat, latlon2.lon, latlon2.lat)
+    azi_angle = az12 + add_angle
+
+    return latlon2.course(azi_angle, radius)
+
+def plot_satellite_track(ax, lon, lat, c, ls, label):
+
+    ax.plot(lon, lat, c=c, ls=ls, label=label)
+
+    for i in range(1, len(lon), 10):  # Adjust the step to control the number of arrows
+        ax.annotate('', xy=(lon[i], lat[i]), xytext=(lon[i-1], lat[i-1]),
+                    arrowprops=dict(arrowstyle="->", color=c))
+
+def plot_forcast_overlay(fig, ax, ifs_ds, ifs_fcst_time, title_label, plot_func, plot_func_kwargs):
+    from datetime import datetime
+
+    plot_func(fig, ax, ifs_ds, **plot_func_kwargs)
+
+    ifs_fcst_time_str = ifs_fcst_time.astype(datetime).strftime("%Y-%m-%d %H:%M:%S")
+    title = ax.get_title()+f"\n({title_label} forecasted on {ifs_fcst_time_str})"
+    ax.set_title(f"{title}", fontsize=15, y=1.1)
+
+def plot_atc_zones(ax, worldfirs_json):
+    import topojson
+    import json
+    import cartopy.crs as ccrs
+
+    with open(worldfirs_json, 'r') as f:
+        data = json.load(f)
+
+    # parse topojson file using `object_name`
+    topo = topojson.Topology(data, object_name="data")
+    firs = topo.to_gdf()
+    kwargs = {
+        "color": 'k',
+        "linestyle": 'dashed',
+        "linewidth": 0.3
+    }
+    for i, row in firs.iterrows():
+        for geom in getattr(row.geometry.boundary, "geoms", [row.geometry.boundary]):
+            lon, lat = zip(*list(geom.coords))
+            if i == 0:
+                kwargs["label"] = "ATC zones"
+                kwargs["alpha"] = 1.0
+            else:
+                kwargs["label"] = None
+                kwargs["alpha"] = 0.4
+            ax.plot(lon, lat, transform=ccrs.PlateCarree(), **kwargs)
+
+def plot_flight_plan_satellite_forecast(figsize,
+                                        flight_time,
+                                        plan,
+                                        domain_lonlat,
+                                        is_ec_track=False,
+                                        ec_track=None,
```

Comment:
[discussed in person] Yes I completely agree but for an unknown reason this didn't work for me because sometimes ec_track was given and yet it was recognised as "is None" (I've no idea why so I just made this weak but quick fix)